community/kernel/src/main/java/org/neo4j/kernel/impl/util/RelationshipProxyWrappingEdgeValue.java

```diff
@@ -107,6 +108,20 @@ public NodeValue endNode()
     }
 
     @Override
+    public NodeValue otherNode( NodeValue node )
+    {
+        if ( node instanceof NodeProxyWrappingNodeValue )
+        {
+            Node proxy = ((NodeProxyWrappingNodeValue) node).nodeProxy();
+            return ValueUtils.fromNodeProxy( relationship.getOtherNode( proxy ) );
+        }
+        else
+        {
+           return super.otherNode( node );
+        }
+    }
```

Comment:
Or is this it? Curious.

Reply:
No it is the line above that calls down to `proxy.getOtherNode` that removes the regression.

Reply:
Awesome.